components/service/glean/src/main/java/mozilla/components/service/glean/private/LabeledMetricType.kt

```diff
@@ -37,7 +37,7 @@ data class LabeledMetricType<T>(
     companion object {
         private const val MAX_LABELS = 16
         private const val OTHER_LABEL = "__other__"
-        private val labelRegex = Regex("^[a-z_][a-z0-9_]{0,29}$")
+        private val labelRegex = Regex("^[a-z_][a-z0-9_]{0,29}(\\\\.[a-z0-9_]{0,29})*\$")
```

Comment:
That very last `\` seems fishy to me.  It's escaping the `$` which *maybe* we need to do since `$` has special meaning in Kotlin strings, but it wasn't escaped before...

Maybe to be sure about this and other reasons, we should add a test [around here](https://github.com/mozilla-mobile/android-components/blob/master/components/service/glean/src/test/java/mozilla/components/service/glean/private/LabeledMetricTypeTest.kt#L256) that sets a label with some dots in it and confirms it gets recorded?

Reply:
The IDE actually inserted that in there when I copied the string verbatim from the glean_parser changes I just made.  I trusted it knew what it was doing and left it there.  I know that the `$` can denote an interpolated string so I'm guessing that is why it was escaped.  I was planning on adding some test cases around this already, which is why I left this as a Draft PR while I was working on the tests.

Reply:
Turns out that _all_ of the escaping that the IDE so _thoughtfully_ added for me was incorrect.  I had to remove the escaped `$` as well as the escaped `\\` in order to get this to work.  So literal of `"^[a-z_][a-z0-9_]{0,29}(\\.[a-z0-9_]{0,29})*$"` was what it needed.

Reply:
:musical_note: Sometimes help is the kind of help, we all could do without :musical_note:

Reply:
This regex makes it unbounded. There's no length limit enforced anymore at all. `*` will _always_ match.

Reply:
Yes, but the total length limit is enforced elsewhere in code.  We still have the benefit that any one section has a max of 29 characters.